Opsero Electronic Design Inc. 2020
Jeff Johnson
ZCU104 FSBL patch 2020.2
========================
The FSBL for Zynq Ultrascale+ needs a patch to properly enable VADJ on the ZCU104 board in the 2020.2
version of PetaLinux. The FSBL released with this version of PetaLinux has code to read the FMC card's
EEPROM and then enable VADJ to the correct value. The released FSBL code in fact reads from the ZCU104
board's EEPROM (address 0x54) and not the FMC's EEPROM (address 0x50). It also only reads 32 bytes 
from the EEPROM, which is not sufficient to include the VADJ voltage data. For both of these reasons, 
the FSBL does not properly enable VADJ on this board if this patch is not applied.

diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.c b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.c
index 3b52900317..6b47d65b15 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.c
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.c
@@ -78,7 +78,7 @@ static u32 XFsbl_ReadMinMaxEepromVadj(XIicPs* I2c0InstancePtr, u32 *MinVadj, u32
 	u32 UStatus;
 	s32 Status;
 	u32 NominalVoltage;
-	u32 EepromAddr = 0x54U;
+	u32 EepromAddr = 0x50U;
 	u32 MinVoltage;
 	u32 MaxVoltage;
 
@@ -87,8 +87,8 @@ static u32 XFsbl_ReadMinMaxEepromVadj(XIicPs* I2c0InstancePtr, u32 *MinVadj, u32
 	MaxVoltage = 0U;
 	XRecord.VadjRecordFound = 0U;
 
-	/* Select the Channel-1 of MUX for I2C EEprom Access */
-	WriteBuffer[0U] = 0x1U;
+	/* Select the Channel-6 of MUX for FMC I2C EEprom Access */
+	WriteBuffer[0U] = 0x20;
 	Status = XIicPs_MasterSendPolled(I2c0InstancePtr,
 				WriteBuffer, 1U, TCA9548A_ADDR);
 	if (Status != XST_SUCCESS) {
diff --git a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.h b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.h
index e6f6684075..b494c47986 100644
--- a/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.h
+++ b/lib/sw_apps/zynqmp_fsbl/src/xfsbl_board.h
@@ -56,7 +56,7 @@ extern "C" {
 #define IRPS5401_SWC_ADDR		0x45U
 
 #define BUF_LEN		10U
-#define MAX_SIZE		32U
+#define MAX_SIZE		256U
 
 #define CMD_CFG_0_REG		0x06U
 #define CMD_OUTPUT_0_REG	0x02U
